### PR TITLE
[FLINK-18224][docs] Add document about sql client's tableau result mode

### PR DESCRIPTION
**Pull request by KurtYoung**
## What is the purpose of the change

Add document about sql client's tableau result mode.

### CODE REVIEW

File: docs/dev/table/sqlClient.md [hunk 1]
```diff
@@ -106,7 +113,33 @@ Alice, 1
 Greg, 1
 {% endhighlight %}
 
-Both result modes can be useful during the prototyping of SQL queries. In both modes, results are stored in the Java heap memory of the SQL Client. In order to keep the CLI interface responsive, the changelog mode only shows the latest 1000 changes. The table mode allows for navigating through bigger results that are only limited by the available main memory and the configured [maximum number of rows](sqlClient.html#configuration) (`max-table-result-rows`).
+In *tableau mode*, if you ran the query in streaming mode, the displayed result would be:
```
Comment (lirui-apache) [inline docs/dev/table/sqlClient.md:116]: Shall we mention how to cancel a job in streaming mode?

File: docs/dev/table/sqlClient.md [hunk 2]
```diff
@@ -106,7 +113,33 @@ Alice, 1
...
+| Alice |   1 |
+|   Bob |   2 |
+|  Greg |   1 |
++-------+-----+
+3 rows in set
+{% endhighlight %}
+
+All these result modes can be useful during the prototyping of SQL queries. In all these modes, results are stored in the Java heap memory of the SQL Client. In order to keep the CLI interface responsive, the changelog mode only shows the latest 1000 changes. The table mode allows for navigating through bigger results that are only limited by the available main memory and the configured [maximum number of rows](sqlClient.html#configuration) (`max-table-result-rows`).
 
 <span class="label label-danger">Attention</span> Queries that are executed in a batch environment, can only be retrieved using the `table` result mode.
```
Comment (godfreyhe) [inline docs/dev/table/sqlClient.md:144]: This line should also be updated: Queries that are executed in a batch environment, can only be retrieved using the `table` result mode and `tableau` result mode.

File: docs/dev/table/sqlClient.md (hunk 1 above)
Comment (JingsongLi) [reply docs/dev/table/sqlClient.md:116]: +1 both batch and streaming mode

File: docs/dev/table/sqlClient.md [hunk 3]
```diff
@@ -106,7 +113,33 @@ Alice, 1
...
+|  name | cnt |
++-------+-----+
+| Alice |   1 |
+|   Bob |   2 |
+|  Greg |   1 |
++-------+-----+
+3 rows in set
+{% endhighlight %}
+
+All these result modes can be useful during the prototyping of SQL queries. In all these modes, results are stored in the Java heap memory of the SQL Client. In order to keep the CLI interface responsive, the changelog mode only shows the latest 1000 changes. The table mode allows for navigating through bigger results that are only limited by the available main memory and the configured [maximum number of rows](sqlClient.html#configuration) (`max-table-result-rows`).
```
Comment (godfreyhe) [inline docs/dev/table/sqlClient.md:142]: `tableau` mode needs be explained more here ?

Comment (KurtYoung) [reply docs/dev/table/sqlClient.md:142]: I don't think this is necessary

File: docs/dev/table/sqlClient.md (hunk 1 above)
Comment (KurtYoung) [reply docs/dev/table/sqlClient.md:120]: added above

**Pull request merged**